samples/java-eventsourced-shopping-cart/src/main/java/com/example/shoppingcart/domain/ShoppingCartTestKit.java

```diff
@@ -0,0 +1,62 @@
+package com.example.shoppingcart;
+
+import com.example.shoppingcart.domain.ShoppingCartDomain;
+import com.example.shoppingcart.ShoppingCartApi;
+import com.google.protobuf.Empty;
+import java.util.ArrayList;
+import java.util.List;
+
+// Q Would we need ShoppingCartImpl as parameter constructor? 
+public class ShoppingCartTestKit {
+
+    ShoppingCartDomain.Cart currentState = ShoppingCartDomain.Cart.newBuilder().build();
+
+    // Q What about having many types of events?
+    public Result<ShoppingCartDomain.ItemAdded> addItem(ShoppingCartApi.AddLineItem command) {
+        // TODO >> change this fake for actual call
+        ShoppingCartDomain.ItemAdded expectedEvent = ShoppingCartDomain.ItemAdded.newBuilder()
+                .setItem(ShoppingCartDomain.LineItem.newBuilder().setProductId(command.getProductId())
+                        .setName(command.getName()).setQuantity(command.getQuantity()).build())
+                .build();
+        // << TODO
+        List<ShoppingCartDomain.ItemAdded> events = new ArrayList<ShoppingCartDomain.ItemAdded>();
+        events.add(expectedEvent);
+        return new Result(Empty.getDefaultInstance(), events, currentState);
+    };
+
+    public Result<ShoppingCartDomain.ItemRemoved> removeItem(ShoppingCartApi.RemoveLineItem command) {
+        return new Result(Empty.getDefaultInstance(), new ArrayList<ShoppingCartDomain.ItemRemoved>(), currentState);
+    };
+
+    // Q or get state?
+    // Q what about the events in this one? what type?
+    public Result getCart(ShoppingCartApi.GetShoppingCart command) {
+        return new Result(Empty.getDefaultInstance(), new ArrayList<ShoppingCartDomain.ItemRemoved>(), currentState);
+    };
+
+    public final class Result<Event> {
```

Comment:
This won't have to be generated but can be a generic `EventSourcedEntityCommandResult<S>`, the event type we do not know, no common supertype with protobuf so will have to be `List<Object>` and then nice methods to inspect/assert

Reply:
Good point. Doesn't need to be generated. What about `com.google.protobuf.GeneratedMessageV3` instead of `Object`?

Reply:
I don't think that has much value to the user, it's more of an implementation detail.

Reply:
I see what you mean, those nice methods to inspect/assert you mention should suffice. The user doesn't gain any type checking with dealing with `GeneratedMessageV3` but with the methods.